framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -1177,6 +1185,11 @@ public void addAnnotation(AnnotationMirror a) {
          *     excluding the receiver
          */
         /*package-private*/ void setParameterTypes(List<AnnotatedTypeMirror> params) {
+            // Record varargsType when initializing paramTypes
+            if (isVarArgs() && paramTypes == null) {
```

Comment:
It was confusing to me why we needed the `paramTypes == null` check. It seems there are two stages:
1. Initialize `paramTypes` to be the actually declared types without adapting parameters.
2. Reset `paramTypes` to adapt parameters.

If that's true, we should probably document this behaviour, and I'm not sure if setting `paramTypesComputed = true` is right in stage 1.